x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/logging/DeprecationIndexingComponent.java

```diff
@@ -110,6 +115,8 @@ public void clusterChanged(ClusterChangedEvent event) {
             }
             appender.setEnabled(newEnabled);
         }
+        final boolean useXOpaqueId = USE_X_OPAQUE_ID_IN_FILTERING.get(state.getMetadata().settings());
```

Comment:
we can consider this to be static setting. We could then avoid using volatile field in RateLimitingFilter

Reply:
This will end up setting the value on every cluster state update. Instead, let's use a settings update consumer?